sdk-core/transaction-intents-factories/token_transfers_factory.md

```diff
@@ -1 +1,28 @@
-// TBD: will be defined in a future PR.
+## TransferIntentsFactory
+
+A class that provides methods for creating transactions for transfers (native or ESDT).
+
+```
+class TransferIntentsFactory:
+    // The constructor is not captured by the specs; it's up to the implementing library to define it.
+    // Generally speaking, the constructor should be parametrized with a configuration object which defines entries such as:
+    // "minGasLimit", "gasLimitPerByte", "issueCost", gas limit for specific operations etc. (e.g. "gasLimitForESDTTransfer").
+    
+    create_transaction_intent_for_native_transfer({
+        sender: IAddress;
+        receiver: IAddress;
+        native_transfer_amount: Amount;
+    }): TransactionIntent;
+
+    // Can throw:
+    // - ErrBadArguments
+    //
+    // All kinds of transfers (native and ESDT) should be handled.
+    // If multiple transfers are specified, a multi-ESDT transfer intent should be created.
+    // Bad usage should be reported.
+    create_transaction_intent_for_custom_transfers({
```

Comment:
question: how does the custom transfer & SC calls are handled? We call first this method and then work with the `TransactionIntent` to add the extra function & parameters? Do we have a factory for that so we should not go in details?

Reply:
For transfer & execute, the developer should use `SmartContractTransactionIntentsFactory` and pass the parameters (as needed):

```
native_transfer_amount: Amount = 0;
custom_transfers: List[CustomTokenTransfer] = [];
```

That factory is responsible with correctly composing the `tx.data`.